usl/usl_kevm.h

```diff
@@ -0,0 +1,241 @@
+#ifndef USL_KEVM_H
+#define USL_KEVM_H
+
+#include <cstdint>
+#include <ctime>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace usl_kevm {
+
+// Common data types
+using bytes_t = std::string;
+using nonce_t = std::uint64_t;
+using ether_t = std::uint64_t;
+using gas_t = std::uint64_t;
+using hash_t = std::uint64_t;
+using account_address_t = std::uint64_t;
+using storage_key_t = std::uint64_t;
+using storage_value_t = std::uint64_t;
+using chain_id_t = std::uint64_t;
+using message_id_t = std::uint64_t;
+using number_t = std::uint64_t;
+
+// Ommer List
+struct ommer_t {
+  account_address_t ommer_address;
+  number_t ommer_number;
+};
+using ommer_list_t = std::vector<ommer_t>;
+
+// Block
+using difficulty_t = std::uint64_t;
+using timestamp_t = std::time_t;
+using mix_hash_t = std::uint64_t;
+struct block_t {
+  hash_t previous_hash;
+  hash_t ommers_hash;
+  account_address_t coinbase;
+  hash_t state_root;
+  hash_t transactions_root;
+  hash_t receipts_root;
+  bytes_t logs_bloom;
+  difficulty_t difficulty;
+  number_t number;
+  gas_t gas_limit;
+  gas_t gas_used;
+  timestamp_t timestamp;
+  bytes_t extra_data;
+  mix_hash_t mix_hash;
+  nonce_t block_nonce;
```

Comment:
I believe this "nonce" is different from an account nonce. You should double check the bitwidth here. Account nonces are 64 bits but I don't know for sure this is the same.

Reply:
I separated this into two types: `account_nonce_t` which according to the yellow paper is a scalar (ie unbounded width natural number) and `block_nonce_t` which according to the yellow paper is 64-bit.

Reply:
It seems that account nonces are also `u64` so I changed the comment to reflect that.